qemu/tests/fio_windows.py

```diff
@@ -0,0 +1,77 @@
+import re
+import time
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+
+from avocado.core import exceptions
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM guest stop test:
+    1) Log into a guest
+    2) Check is fio.msi installed, install it if not installed.
+    3) Start fio in guest
+    4) Check vm is alive
+    5) Get the result
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    install_path = params["install_path"].rstrip("\\")
+    log_file = params.get("log_file")
+    start_cmd = params.get("start_cmd")
+    check_running_cmd = "tasklist|findstr /I fio"
+    check_running_cmd = params.get("check_running_cmd", check_running_cmd)
+    check_installed_cmd = 'dir "%s"|findstr /I fio' % install_path
+    check_installed_cmd = params.get("check_installed_cmd",
+                                     check_installed_cmd)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+    # Start another session with the guest and
+    # make sure the background process is running
+    session2 = vm.wait_for_login(timeout=timeout)
+    try:
+        installed = session.cmd_status(check_installed_cmd) == 0
+        if not installed:
+            dst = r"%s:\\" % utils_misc.get_winutils_vol(session)
+
+            error_context.context("Install fio in guest", logging.info)
+            install_cmd = params["install_cmd"]
+            install_cmd = re.sub(r"DRIVE:\\+", dst, install_cmd)
+            session.cmd(install_cmd, timeout=180)
+            time.sleep(30)
+            config_cmd = params.get("config_cmd")
+            if config_cmd:
+                session.cmd(config_cmd)
+
+        error_context.context("Start fio in guest.", logging.info)
+        start_cmd = params["start_cmd"]
+        test_timeout = int(params.get("test_timeout", "360"))
+        logging.debug("fio cmd: %s" % start_cmd)
+        session2.sendline(start_cmd)
+
+        running = utils_misc.wait_for(lambda: session.cmd_status(
```

Comment:
still suggest to use session.cmd_status_output instead of session2.sendline,

you will lost the failed error if the test not failed at the beginning.

@xutian what's your opinion